src/inference/hmc.jl

```diff
@@ -354,12 +398,23 @@ Arguments:
 - `init_ϵ::Float64` : Inital step size; 0 means automatically searching using a heuristic procedure.
 
 """
-struct NUTS{AD,space,metricT<:AHMC.AbstractMetric} <: AdaptiveHamiltonian{AD}
+struct NUTS{
+    AD,
+    space,
+    metricT <: AHMC.AbstractMetric,
+    TS<:AHMC.AbstractTrajectorySampler,
+    TC<:AHMC.AbstractTerminationCriterion,
+    I<:AHMC.AbstractIntegrator,
+    A<:AHMC.AbstractAdaptor
+} <: AdaptiveHamiltonian{AD}
     n_adapts::Int         # number of samples with adaption for ϵ
     δ::Float64        # target accept rate
     max_depth::Int         # maximum tree depth
     Δ_max::Float64
     ϵ::Float64     # (initial) step size
+    metric::metricT
+    integrator::I
```

Comment:
A minor comment: `integrator` seems a bit redundant with `ϵ` here. Maybe consider removing the `ϵ` field in this type, and storing the information in `DefaultIntegrator` when a user specifies `ϵ` in the constructor?